xdis/code.py

```diff
@@ -637,3 +640,63 @@ def code_has_star_star_arg(code):
     """Return True iff
     The code object has a variable keyword parameter (**kwargs-like)."""
     return (code.co_flags & 8) != 0
+
+
+# From: https://stackoverflow.com/questions/36330860/pythonically-check-if-a-variable-name-is-valid
+def is_valid_variable_name(name):
+    """Returns True iff
+    the argument is a valid Python variable name"""
+    if not re.fullmatch('^[_a-zA-Z][_0-9a-zA-Z]*$', name):
+        return False
+    try:
+        ast.parse('{} = None'.format(name))
+        return True
+    except (SyntaxError, ValueError, TypeError):
+        return False
+
+
+def is_valid_variable_names(code):
+    """Return True iff all of the co_names are valid Python identifier names"""
+    return all(is_valid_variable_name(name) for name in code.co_names)
+
+
+def fix_variable_name(name):
+    """Converts an invalid python variable name into a valid variable name similar to the input"""
+    # Replace invalid character with underscore
+    name = re.sub('[^_0-9a-zA-Z]', '_', name)
+    # Replace leading digit with underscore
+    name = re.sub('^[0-9]', '_', name)
+    if not is_valid_variable_name(name):
+        return '_' + name
+    else:
+        return name
+
+
+def fix_variable_names(code):
+    """Modifies a code object, transforming all invalid names into valid names and avoiding collisions."""
+    valid_names = UniqueSuffixSet()
+    fixed_names = []
+    for co_name in code.co_names:
+        fixed_name = valid_names.add(fix_variable_name(co_name))
+        fixed_names.append(fixed_name)
+
+    args = [code.co_argcount]
+    if PYTHON3:
+        args.append(code.co_kwonlyargcount)
+    args += [
```

Comment:
Recently I've come to learn that Python 3.8 there is now a "replace" method on Code. See https://docs.python.org/3/library/types.html#types.CodeType

So xdis in its code type should add this method (for all types of CodeType) . So the way ths might work to create an xdis code type and then convert that to the local code type.

I realize this is a digression from the main topic here, so it can be deferred. I do want to note it becuase this kind of code is pervasive and ugly and error prone.

Reply:
That sounds good. I'm not super familiar with code objects especially not accross different Python versions and I'm also not familiar with this codebase so I don't know what Python versions xdis is aming to be compatible with but I'll gladly change this part of the code anytime as it is by far the worst part of the whole PR.